lib/datasets.go

```diff
@@ -1009,16 +1009,24 @@ func (m *DatasetMethods) Validate(p *ValidateParams, res *ValidateResponse) erro
 		return qrierr.New(ErrBadArgs, "please provide a dataset name, or a supply the --body and --schema or --structure flags")
 	}
 
-	ref, err := repo.ParseDatasetRef(p.Ref)
-	if err != nil && err != repo.ErrEmptyRef {
-		return err
-	}
-	err = repo.CanonicalizeDatasetRef(m.inst.repo, &ref)
-	if err != nil && err != repo.ErrEmptyRef {
-		if err == repo.ErrNotFound {
-			return fmt.Errorf("cannot find dataset: %s", ref)
+	fsiPath := ""
+	var err error
+	ref := dsref.Ref{}
+
+	// if there is both a bodyfilename and a schema/structure
+	// we don't need to resolve any references
+	if !(p.BodyFilename != "" && schemaFlagType != "") {
+		// TODO (ramfox): we need consts in `dsref` for "local", "network", "p2p"
+		ref, _, err = m.inst.ParseAndResolveRefWithWorkingDir(ctx, p.Ref, "local")
+		if err != nil {
+			return err
+		}
+
+		// TODO (b5) - replace this prefix check with a call to qfs.PathKind when it
+		// supports the fsi prefix
```

Comment:
Would `qfs` need anything more than just adding `fsi` to the `PathKind` func? Would we have to guarantee any other `fsi` functionality?

Reply:
Would love to have this as a common utility function, like `fsi.IsFSIPath` so that we don't expose the hard-coded string constant everywhere. I don't know the specific plan for `qfs`, but having a utility function for now would make an eventual switch easier.